src/structure/spatial.jl

```diff
@@ -222,6 +253,103 @@ function psiangle(res::AbstractResidue, res_next::AbstractResidue)
     return dihedralangle(res["N"], res["CA"], res["C"], res_next["N"])
 end
 
+function psiangle(chain::Chain, res_id::Union{Integer, String})
+    inds = find(r -> resid(r)==string(res_id), chain)
+    if length(inds) != 1
+        throw(ArgumentError("$res_id is an invalid residue ID"))
+    end
+    res_list = collect(chain)
+    i = inds[1]
+    if i == length(res_list) || !sequentialresidues(res_list[i], res_list[i + 1])
+        throw(ArgumentError("$res_id is an invalid residue ID"))
+    end
+    return psiangle(res_list[i], res_list[i + 1])
+end
+
+"""
+Calculate vector of omega, psi or phi angles for a `StructuralElementOrList`.
+"""
+function omegaangles(el::StructuralElementOrList)
```

Comment:
I think to be more flexible the user should be able to pass any selector. So the arguments and `collectresidues` line should be the same as `ramachandranangles`.